utils_cv/tracking/model.py

```diff
@@ -243,9 +279,48 @@ def save(self, path) -> None:
         save_model(path, self.epoch, self.model, self.optimizer)
         print(f"Model saved to {path}")
 
-    def evaluate(self, results, gt) -> pd.DataFrame:
-        pass
+    def evaluate(self,
+                 results: Dict[int, List[TrackingBbox]],
+                 gt_root_path: str) -> pd.DataFrame:
+        
+        """ eval code that calls on 'motmetrics' package in referenced FairMOT script, to produce MOT metrics on inference, given ground-truth.
+        Args:
+            results: prediction results from predict() function, i.e. Dict[int, List[TrackingBbox]] 
+            gt_root_path: path of dataset containing GT annotations in MOTchallenge format (xywh)
+        Returns:
+            strsummary: pandas.DataFrame output by method in 'motmetrics', containing metrics scores
+        Raises:
+            Exception: if both `results` and `self.stored_predictions` are None.
+        """
+        if results is None:
+            if not self.stored_predictions: #TODO: add stored_predictions in predict() 
+                raise Exception("No predict() function run on dataset for for evaluation")
+            results = self.stored_predictions     
+        
+        result_filename = osp.join(gt_root_path,'results', 'results.txt')
```

Comment:
Instead of writing this file to the user directory, use tempfile.TemporaryDirectory() instead. You can then create a subdirectory called "results" there which the user will never see.